app/scripts/modules/core/src/pipeline/config/triggers/PipelineRoles.tsx

```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
+import Select, { Option } from 'react-select';
+
+import { AuthenticationService } from 'core/authentication';
+import { HelpField } from 'core/help';
+
+const { useState, useEffect } = React;
+
+export interface IPipelineRolesConfigProps {
+  roles: any[];
+  updateRoles: (roles: string[]) => void;
+}
+
+export const PipelineRoles = (props: IPipelineRolesConfigProps) => {
+  const [allowedRoles, setAllowedRoles] = useState([]);
+
+  useEffect(() => {
```

Comment:
I don't think you need a `useEffect` at all...  since getAuthenticatedUser() is synchronous you can either:

1) simply grab the value each time
2) cache it using `useMemo`
```const allowedRoles = useMemo(() => AuthenticationService.getAuthenticatedUser().roles, []);```